src/sbvr-api/control-flow.ts

```diff
@@ -17,12 +17,23 @@ export const liftP = <T, U>(fn: (v: T) => Resolvable<U>) => {
 export type MappingFunction = <T, U>(
 	a: T[],
 	fn: (v: T) => Resolvable<U>,
-) => Bluebird<Array<U | Error>>;
+) => Promise<Array<U | Error>>;
 
 // The settle version of `Promise.mapSeries`
-export const settleMapSeries: MappingFunction = (a, fn) => {
-	const runF = Bluebird.method(fn);
-	return Bluebird.mapSeries(a, _.flow(runF, wrap));
+export const settleMapSeries: MappingFunction = async <T, U>(
+	a: T[],
+	fn: (v: T) => Resolvable<U>,
+) => {
```

Comment:
How come you have to declare types both for the var and the arguments? Why isn't `MappingFunction` enough for TS to infer the argument types?

Reply:
It's enough for it to infer the types but we won't have access to `U` for the `const results: Array<U | Error> = [];` below